api_app/analyzers_manager/file_analyzers/mobsf_service.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class MobSF_Service(FileAnalyzer):
+    mobsf_host: str
+    _mobsf_api_key: str
+
+    def update(self) -> bool:
+        pass
+
+    def static_analysis(self, scan_hash, headers):
+
+        scan_url = self.mobsf_host + "/api/v1/scan"
+        data = {"hash": scan_hash}
+        scan_response = requests.post(url=scan_url, data=data, headers=headers)
+        scan_response.raise_for_status()
+
+        report_url = self.mobsf_host + "/api/v1/report_json"
+        report_response = requests.post(url=report_url, data=data, headers=headers)
+        report_response.raise_for_status()
+        return report_response.json()
+
+    def dynamic_analysis(self, scan_hash, headers):
+
+        mobsfy_runtime_response = requests.post(
+            url=self.mobsf_host + "/api/v1/android/mobsfy",
+            headers=headers,
+            data={"identifier": "127.0.0.1:6555"},
+        )
+        mobsfy_runtime_response.raise_for_status()
+
+        start_dynamic_analysis_response = requests.post(
+            url=self.mobsf_host + "/api/v1/dynamic/start_analysis",
+            headers=headers,
+            data={"hash": scan_hash},
+        )
+        start_dynamic_analysis_response.raise_for_status()
+
+        tls_tests_response = requests.post(
+            url=self.mobsf_host + "/api/v1/android/tls_tests",
+            headers=headers,
+            data={"hash": scan_hash},
+        )
+        tls_tests_response.raise_for_status()
+
+        start_frida_instrumentation_response = requests.post(
+            url=self.mobsf_host + "/api/v1/frida/instrument",
+            headers=headers,
+            data={
+                "hash": scan_hash,
+                "default_hooks": "api_monitor,ssl_pinning_bypass,root_bypass,debugger_check_bypass",
+                "auxiliary_hooks": "",
+                "frida_code": "",
+            },
+        )
+        start_frida_instrumentation_response.raise_for_status()
+
+        get_runtime_dependency_response = requests.post(
+            url=self.mobsf_host + "/api/v1/frida/get_dependencies",
+            headers=headers,
+            data={"hash": scan_hash},
+        )
+        get_runtime_dependency_response.raise_for_status()
+
+        stop_dynamic_analysis = requests.post(
+            url=self.mobsf_host + "/api/v1/dynamic/stop_analysis",
```

Comment:
as you suggested, dynamic analysis time is debatable. This is the reason why I think we should add here a sleep and that this should be a parameter of the analyzer. Which default? 20-30 seconds? What do you think about?

Reply:
As I commented earlier, in static analysis review comment, wouldn't just configuring the soft-time limit do the trick. Because if I keep it at default value, "SoftTimeLimitExceeded" error is thrown and my job fails. I had to set it to 400 in order for my job to be successful. Let me know what you think!

Reply:
see previous comment

Reply:
I noticed that no sleep was added so basically the analysis has a really brief duration time, just the time needed to execute all the steps between the calls "start" and "stop". As mentioned, I would add a sleep between the "start" and the later calls, to allow time for the stuff to start properly and the code loaded and maybe to allow the malware to do some nasty things too.